src/lib/pages/validators/index.tsx

```diff
@@ -0,0 +1,150 @@
+import { Flex, Heading, Text } from "@chakra-ui/react";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
+import { AmpEvent, track } from "lib/amplitude";
+import { useMobile } from "lib/app-provider";
+import InputWithIcon from "lib/components/InputWithIcon";
+import PageContainer from "lib/components/PageContainer";
+import { Pagination } from "lib/components/pagination";
+import { usePaginator } from "lib/components/pagination/usePaginator";
+import { useDebounce } from "lib/hooks";
+import { useValidators } from "lib/services/validatorService";
+
+import { ActiveFilter, OrderSelect } from "./components";
+import { ValidatorsTable } from "./components/validators-table";
+import { ValidatorOrder } from "./types";
+
+const Validators = () => {
+  const router = useRouter();
+  const isMobile = useMobile();
+
+  const [isActive, setIsActive] = useState(true);
+  const [order, setOrder] = useState(ValidatorOrder.VotingPower);
+  const [isDesc, setIsDesc] = useState(true);
+  const [search, setSearch] = useState("");
+  const debouncedSearch = useDebounce(search);
+
+  const {
+    pagesQuantity,
+    setTotalData,
+    currentPage,
+    setCurrentPage,
+    pageSize,
+    setPageSize,
+    offset,
+  } = usePaginator({
+    initialState: {
+      pageSize: 100,
+      currentPage: 1,
+      isDisabled: false,
+    },
+  });
+  const { data, isFetching: isLoading } = useValidators(
+    pageSize,
+    offset,
+    isActive,
+    order,
+    isDesc,
+    debouncedSearch,
+    {
+      onSuccess: ({ total }) => setTotalData(total),
+    }
+  );
+
+  useEffect(() => {
+    if (router.isReady) track(AmpEvent.TO_VALIDATORS);
+  }, [router.isReady]);
+
+  useEffect(() => {
+    setCurrentPage(1);
+    setPageSize(100);
+  }, [isActive, order, isDesc, debouncedSearch, setCurrentPage, setPageSize]);
```

Comment:
chainId?

Reply:
chainId is already handled in `usePaginator`